pkg/basinprovider/provider.go

```diff
@@ -165,12 +172,16 @@ func (bp *BasinProvider) Upload(
 		}
 		signer.Sum(buf)
 
-		err = callback.Write(ctx, func(p Publications_Callback_write_Params) error {
+		f, release := callback.Write(ctx, func(p Publications_Callback_write_Params) error {
 			return p.SetChunk(buf)
 		})
+		defer release()
 		if err != nil {
```

Comment:
looks like this `err` is already checked above?

Reply:
leftover from the old call. thanks!